spec/services/s3_service_spec.rb

```diff
@@ -73,4 +73,10 @@
     remote_path = "originals/#{child_object_oid}.tif"
     expect(described_class.s3_exists?(remote_path)).to eq false
   end
+
+  it "can generate signed URL" do
+    child_object_oid = "1014543"
+    remote_path = "originals/#{child_object_oid}.tif"
+    expect(described_class.presigned_url(remote_path, 600)).to start_with "https://#{ENV['S3_SOURCE_BUCKET_NAME']}"
```

Comment:
I added some more expects for the URL returned to make sure it seems to match the remote_path the the expiration parameter is correct.  Also added a test to indicate that the returned value is a String.
The URL goes to S3, so I could try downloading the file and see if I can get it...check headers etc.  But, I wonder if that's testing S3 more than the presigning.  Take a look at the updated test (and the comment for the presigned_url) and let me know if you think it should actually make the HTTP request to get the image from S3 using the URL.

Reply:
Maybe it would work to add the check in the system spec for child object (still to be done).  It could make sure the image is displayed there, which would be testing the presigning as an system test.